sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -245,11 +252,15 @@ case class HashAggregateExec(
        |   $doAggFuncName();
        |   $aggTime.add((System.nanoTime() - $beforeAgg) / 1000000);
        |
-       |   // output the result
-       |   ${genResult.trim}
+       |   if (!$hasInput && ${resultVars.isEmpty}) {
```

Comment:
We only need this check only if `resultExpressions` is empty. So, I think we can drop this check if non-empty (and regular) cases.

Reply:
I think it hurts the code readability if the code for the two cases are defined separately.  For the regular case, the generated code will look like `if (false && !hasInput) ... else ...`. This pattern should be optimized easily during jit, so we don't need to worry about the performance too much.